packages/hydrogen/src/storefront.ts

```diff
@@ -368,15 +368,15 @@ export function createStorefrontClient<TI18n extends I18nBase>(
 
     const {data, errors} = body as GraphQLApiResponse<T>;
 
-    if (errors?.length) {
+    if (!data && errors?.length) {
       throwGraphQLError({
         ...errorOptions,
         errors,
         ErrorConstructor: StorefrontApiError,
       });
     }
 
-    return data as T;
+    return {...data, errors} as T & {errors?: unknown[]};
```

Comment:
Agree with @frandiox. 

Personally, I think the user should just get errors and not two types of errors back. I would standardize both to just `errors` and make sure that "apiErrors" have the same type as regular Graphql error, perhaps with text prefix before the error that helps them be identified as api-specific errors.

Reply:
The reason why I left this named as `errors` is because it's part of [gql standard response format](https://spec.graphql.org/October2021/#sec-Errors).

`storefront.query` and `storefront.mutate` is consider raw fetch for storefront api, it would kinda weird if we change the output format.

The functions of `createCartHandler` is an abstraction. In order to be not confused by `errors` returned by cart mutations which have type `CartUserError | MetafieldsSetUserError | MetafieldDeleteUserError`, I give it a name `apiErrors` .. maybe it should be `gqlErrors` to be even more precise on what it is.